google-api-client 1.29.2

google-api-client 1.30.2 is the latest release of google-api-client. 1.29.2 is the latest without changing artifact ID of google-http-client-jackson

![image](https://user-images.githubusercontent.com/28604/62792702-cfc5f680-ba9d-11e9-92eb-ee35a11caea3.png)


I tried to upgrade the version to 1.30.2 but encountered 2 missing dependencies:
```
[ERROR] Failed to execute goal on project datastore-v1-proto-client: Could not resolve dependencies for project com.google.cloud.datastore:datastore-v1-proto-client:jar:1.6.1: The following artifacts could not be resolved:
 com.google.http-client:google-http-client-jackson:jar:1.30.2,
 com.google.oauth-client:google-oauth-client:jar:1.30.2:
Could not find artifact com.google.http-client:google-http-client-jackson:jar:1.30.2 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

--- a/java/datastore/pom.xml
+++ b/java/datastore/pom.xml
@@ -50,7 +50,7 @@
 
     <dependency>
       <groupId>com.google.http-client</groupId>
-      <artifactId>google-http-client-jackson</artifactId>
+      <artifactId>google-http-client-jackson2</artifactId>
     </dependency>
 
     <dependency>

--- a/java/datastore/src/main/java/com/google/datastore/v1/client/DatastoreHelper.java
+++ b/java/datastore/src/main/java/com/google/datastore/v1/client/DatastoreHelper.java
@@ -22,7 +22,7 @@ import com.google.api.client.http.GenericUrl;
 import com.google.api.client.http.HttpRequest;
 import com.google.api.client.http.HttpTransport;
 import com.google.api.client.json.JsonFactory;
-import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
 import com.google.datastore.v1.ArrayValue;
 import com.google.datastore.v1.CompositeFilter;
 import com.google.datastore.v1.Entity;

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -70,12 +70,6 @@
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client</artifactId>
         <version>1.32.1</version>
-        <exclusions>
-          <exclusion>
-            <artifactId>guava-jdk5</artifactId>
-            <groupId>com.google.guava</groupId>
-          </exclusion>
-        </exclusions>
       </dependency>
 
       <dependency>
@@ -87,19 +81,19 @@
       <dependency>
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client-jackson</artifactId>
-        <version>1.32.1</version>
+        <version>1.29.2</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.oauth-client</groupId>
         <artifactId>google-oauth-client</artifactId>
-        <version>1.32.1</version>
+        <version>1.30.4</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.api-client</groupId>
         <artifactId>google-api-client</artifactId>
-        <version>1.32.1</version>
+        <version>1.30.5</version>
       </dependency>
 
       <dependency>
@@ -122,7 +116,7 @@
       </dependency>
     </dependencies>
   </dependencyManagement>
-  
+
   <build>
     <pluginManagement>
       <plugins>
@@ -184,7 +178,7 @@
             </execution>
           </executions>
         </plugin>
-          
+
       </plugins>
     </pluginManagement>
 

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -80,8 +80,8 @@
 
       <dependency>
         <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client-jackson</artifactId>
-        <version>1.29.2</version>
+        <artifactId>google-http-client-jackson2</artifactId>
+        <version>1.32.1</version>
       </dependency>
 
       <dependency>

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -68,20 +68,10 @@
 
       <dependency>
         <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client</artifactId>
+        <artifactId>google-http-client-bom</artifactId>
         <version>1.32.1</version>
-      </dependency>
-
-      <dependency>
-        <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client-protobuf</artifactId>
-        <version>1.32.1</version>
-      </dependency>
-
-      <dependency>
-        <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client-jackson2</artifactId>
-        <version>1.32.1</version>
+        <type>pom</type>
+        <scope>import</scope>
       </dependency>
 
       <dependency>

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -69,7 +69,7 @@
       <dependency>
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client</artifactId>
-        <version>1.28.0</version>
+        <version>1.32.1</version>
         <exclusions>
           <exclusion>
             <artifactId>guava-jdk5</artifactId>
@@ -81,31 +81,31 @@
       <dependency>
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client-protobuf</artifactId>
-        <version>1.28.0</version>
+        <version>1.32.1</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client-jackson</artifactId>
-        <version>1.28.0</version>
+        <version>1.32.1</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.oauth-client</groupId>
         <artifactId>google-oauth-client</artifactId>
-        <version>1.28.0</version>
+        <version>1.32.1</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.api-client</groupId>
         <artifactId>google-api-client</artifactId>
-        <version>1.28.0</version>
+        <version>1.32.1</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.guava</groupId>
         <artifactId>guava</artifactId>
-        <version>20.0</version>
+        <version>28.1-android</version>
       </dependency>
 
       <dependency>
@@ -117,7 +117,8 @@
       <dependency>
         <groupId>com.google.truth</groupId>
         <artifactId>truth</artifactId>
-        <version>0.42</version>
+        <version>1.0</version>
+        <scope>test</scope>
       </dependency>
     </dependencies>
   </dependencyManagement>

